File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hashMap = {};
    for (let i = 0; i < nums.length; i++) {
        const calc = target - nums[i];
        if (calc in hashMap) {
            return [hashMap[calc], i];
        } else {
            hashMap[nums[i]] = i;
        }
    }
    return [0, 0];
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> previousElements = new();
        for (int i = 0; i < nums.Length; i++) {
            int valueToFind = target - nums[i];
            if (previousElements.ContainsKey(valueToFind)) {
                return new int[] {previousElements[valueToFind], i};
            }
            if (!previousElements.ContainsKey(nums[i])) {
                previousElements.Add(nums[i], i);    
            }
        }
        return [0, 0];
    }
}